x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { TrustedApp } from '../../../../../common/endpoint/types/trusted_apps';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { getTrustedAppsListPath } from '../../../common/routing';
+
+import {
+  getListCurrentPageIndex,
+  getListCurrentPageSize,
+  getListErrorMessage,
+  getListItems,
+  getListTotalItemsCount,
+  isListLoading,
+} from '../store/selectors';
+
+import { useTrustedAppsSelector } from './hooks';
+
+import { FormattedDate } from '../../../../common/components/formatted_date';
+
+const OS_TITLES: { [K in TrustedApp['os']]: string } = {
+  windows: i18n.translate('xpack.securitySolution.trustedapps.os.windows', {
+    defaultMessage: 'Windows',
+  }),
+  macos: i18n.translate('xpack.securitySolution.trustedapps.os.macos', {
+    defaultMessage: 'Mac OS',
+  }),
+  linux: i18n.translate('xpack.securitySolution.trustedapps.os.linux', {
+    defaultMessage: 'Linux',
+  }),
+};
+
+const COLUMN_TITLES: { [K in keyof Omit<TrustedApp, 'id' | 'entries'>]: string } = {
+  name: i18n.translate('xpack.securitySolution.trustedapps.list.columns.name', {
+    defaultMessage: 'Name',
+  }),
+  os: i18n.translate('xpack.securitySolution.trustedapps.list.columns.os', {
+    defaultMessage: 'OS',
+  }),
+  created_at: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdAt', {
+    defaultMessage: 'Date Created',
+  }),
+  created_by: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdBy', {
+    defaultMessage: 'Created By',
+  }),
+};
+
+const COLUMN_DEFINITIONS = [
```

Comment:
Along with Kevin's comment - if teh decision is to leave these here, please make it Immutable by using `Object.freeze()`.

Reply:
Made this to be returned by a function and wrapped into useMemo in component.